sdk/storage/azblob/blockblob/client_test.go

```diff
@@ -587,6 +588,624 @@ func (s *BlockBlobRecordedTestsSuite) TestUploadBlockWithImmutabilityPolicy() {
+		},
+	}
+
+	_, err = destBlob.SetMetadata(context.Background(), nil, nil) // SetMetadata changes the blob's etag
+	_require.Nil(err)
+
+	_, err = destBlob.UploadBlobFromURL(context.Background(), srcBlobURLWithSAS, &options)
+	_require.Nil(err)
+
+	resp, err = destBlob.GetProperties(context.Background(), nil)
+	_require.Nil(err)
+}
+
+func (s *BlockBlobUnrecordedTestsSuite) TestPutBlobFromUrlDestIfNoneMatchFalse() {
+	_require := require.New(s.T())
+	testName := s.T().Name()
+	svcClient, err := testcommon.GetServiceClient(s.T(), testcommon.TestAccountDefault, nil)
+	_require.NoError(err)
+
+	containerClient, _, destBlob, srcBlobURLWithSAS, _ := setUpPutBlobFromURLTest(s, testName, _require, svcClient)
+	defer testcommon.DeleteContainer(context.Background(), _require, containerClient)
+
+	// Get ETag from dest blob
+	resp, err := destBlob.GetProperties(context.Background(), nil)
+	_require.Nil(err)
+
+	// Invoke PutBlobFromURL, should fail and validate error
+	options := blockblob.UploadBlobFromURLOptions{
+		AccessConditions: &blob.AccessConditions{
+			ModifiedAccessConditions: &blob.ModifiedAccessConditions{
+				IfNoneMatch: resp.ETag,
+			},
+		},
+	}
+
+	_, err = destBlob.UploadBlobFromURL(context.Background(), srcBlobURLWithSAS, &options)
+	_require.NotNil(err)
+	testcommon.ValidateBlobErrorCode(_require, err, bloberror.ConditionNotMet)
+}
+
+func (s *BlockBlobUnrecordedTestsSuite) TestPutBlobFromURLCopySourceFalse() {
+	_require := require.New(s.T())
+	testName := s.T().Name()
+	svcClient, err := testcommon.GetServiceClient(s.T(), testcommon.TestAccountDefault, nil)
+	_require.NoError(err)
+
+	containerClient, srcBlob, destBlob, srcBlobURLWithSAS, _ := setUpPutBlobFromURLTest(s, testName, _require, svcClient)
+	defer testcommon.DeleteContainer(context.Background(), _require, containerClient)
+
+	// Set tier to Cool and check tier has been set
+	_, err = srcBlob.SetTier(context.Background(), blob.AccessTierCool, nil)
+	_require.Nil(err)
+
+	resp, err := srcBlob.GetProperties(context.Background(), nil)
+	_require.Nil(err)
+	_require.Equal(resp.AccessTier, to.Ptr("Cool"))
+
+	// Invoke UploadBlobForURL
+	// CopySourceBlobProperties is true by default, trying false here
+	options := blockblob.UploadBlobFromURLOptions{
+		CopySourceBlobProperties: to.Ptr(false),
+	}
+
+	pbResp, err := destBlob.UploadBlobFromURL(context.Background(), srcBlobURLWithSAS, &options)
+	_require.NoError(err)
+	_require.NotNil(pbResp)
+
+	// Access Tier for dest blob will not be Cool
+	resp, err = destBlob.GetProperties(context.Background(), nil)
+	_require.NoError(err)
+	_require.NotEqual(resp.AccessTier, to.Ptr("Cool"))
+}
+
+func (s *BlockBlobRecordedTestsSuite) TestPutBlobFromURLCopySourceAuth() {
+	_require := require.New(s.T())
+	testName := s.T().Name()
+	svcClient, err := testcommon.GetServiceClient(s.T(), testcommon.TestAccountDefault, nil)
+	_require.NoError(err)
+
+	// Getting AAD Authentication
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
```

Comment:
The recording for this test is failing because,

1. There is no environment variable **AZURE_TENANT_ID** in the recording. This environment variable is used by the `azidentity.NewDefaultAzureCredential()` method to return the token credential. You can try a workaround like [this](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/storage/azblob/internal/testcommon/clients_auth.go#L148).
2. `cred.GetToken()` also returns different token strings which is passed in the `x-ms-copy-source-authorization` request header. So, the test fails in playback mode since it is unable to find the recording for the given CopySourceAuthorization header. Maybe something like [this](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/storage/azblob/internal/testcommon/common.go#L185) can help here.